kore/app/exec/Main.hs

```diff
@@ -624,20 +629,27 @@ mainDispatch :: KoreExecOptions -> Main ExitCode
 mainDispatch execOptions
     | Just proveOptions@KoreProveOptions{bmc} <- koreProveOptions =
         if bmc
-            then koreBmc execOptions proveOptions
-            else koreProve execOptions proveOptions
+            then koreBmc execOptions proveOptions & warnProductivity
+            else koreProve execOptions proveOptions & warnProductivity
     | Just searchOptions <- koreSearchOptions =
-        koreSearch execOptions searchOptions
+        koreSearch execOptions searchOptions & warnProductivity
     | Just mergeOptions <- koreMergeOptions =
-        koreMerge execOptions mergeOptions
+        koreMerge execOptions mergeOptions & warnProductivity
     | otherwise =
-        koreRun execOptions
+        koreRun execOptions & warnProductivity
```

Comment:
Would it be better if we had a `mainDispatchWorker` and `mainDispatch` would look like `warnProductivity mainDispatchWorker` or something similar? The idea is that we call `warnProductivity` on every branch now, and we could reduce duplication.